rest-api/src/main/scala/za/co/absa/enceladus/rest_api/services/RunService.scala

```diff
@@ -16,32 +16,31 @@
 package za.co.absa.enceladus.rest_api.services
 
 import java.util.UUID
-
 import com.mongodb.MongoWriteException
 import org.joda.time.format.DateTimeFormat
 import org.springframework.beans.factory.annotation.{Autowired, Value}
 import org.springframework.stereotype.Service
 import za.co.absa.atum.model.{Checkpoint, ControlMeasure, RunStatus}
 import za.co.absa.enceladus.rest_api.exceptions.{NotFoundException, ValidationException}
 import za.co.absa.enceladus.rest_api.models.{RunDatasetNameGroupedSummary, RunDatasetVersionGroupedSummary, RunSummary, TodaysRunsStatistics}
-import za.co.absa.enceladus.rest_api.repositories.RunMongoRepository
+import za.co.absa.enceladus.rest_api.repositories.{MongoRepository, RunMongoRepository}
 import za.co.absa.enceladus.model.{Run, SplineReference, Validation}
 
 import scala.concurrent.Future
 import scala.util.{Failure, Success, Try}
 
-@Service
-class RunService @Autowired()(val mongoRepository: RunMongoRepository)
+@Service("runService") // by-name qualifier: making V2 autowiring un-ambiguous
```

Comment:
If you would have time to explain this to me (educating 😉 )

Reply:
`RunServiceV3` extends `RunService`. This has a few expected effects:
 - Wherever a field of type `RunServiceV3` is needed, `@Autowired` only has 1 option to use - `RunServiceV3`.
 - Wherever a field of type `RunService` is needed, `@Autowired` only has 2 options to use - `RunServiceV3` or `RunServiceV3` and thus fails with ambiguity exception, therefore it needs to be disambigued somehow (some details [here](https://www.baeldung.com/spring-autowire#disambiguation), but basically this is summed up by _By default, Spring resolves @Autowired entries by type._)

There are multiple ways how to hint the desired wiring to be used, e.g. mark one of the implementations by `@Primary` etc. The option I have used here is that I mark the bean with a custom qualifier that matches the field name where requested. So, I have:

```scala
@Service("runService") // by-name qualifier: making V2 autowiring un-ambiguous
class RunService()

@Service
class RunServiceV3()

class LandingPageController @Autowired() (
  ...,
  runService: RunService, // matches the qualifier of RunService => no ambiguity, RunService is wired.
)
```